### PR TITLE
Camera91 zelin

### PR DESCRIPTION
**Pull request by ZhangZelin**
Movement/Camera should be fixed. It gets really buggy when gravity starts affecting the player after a player activates a star but that should be addressed in Issue #93

### CODE REVIEW

File: Assets/Scripts/ChangeScene.cs [hunk 1]
```diff
@@ -6,9 +6,8 @@ public class ChangeScene : MonoBehaviour
 {
     public string sceneName;
 
-    public void ChangeGameScene(string sceneName)
+    public void ChangeGameScedne(string sceneName)
```
Comment (xAmaranth) [inline Assets/Scripts/ChangeScene.cs]: Typo

File: Assets/Scripts/ThirdPersonCamera.cs [hunk 2]
```diff
@@ -90,17 +97,19 @@ private void FixedUpdate()
...
                     ref _currentCameraVelocity,
                     cameraFollowDelay);
             }
-
+            
             if (Math.Abs(xAxisInput) > 0.1f || Math.Abs(yAxisInput) > 0.1f)
             {
-                distanceToTarget = Vector3.Distance(_mainCamera.position, _cameraTarget.position);
+                var position = _mainCamera.position;
+                var position1 = _cameraTarget.position;
```
Comment (xAmaranth) [inline Assets/Scripts/ThirdPersonCamera.cs]: These names suck. 
Maybe mainCameraPos and cameraTargetPos.

File: Assets/Scripts/ThirdPersonPlayer.cs [hunk 3]
```diff
@@ -2,11 +2,17 @@
 using UnityEngine;
 using System.Collections.Generic;
 using TMPro;
+using UnityEngine.Serialization;
```
Comment (xAmaranth) [inline Assets/Scripts/ThirdPersonPlayer.cs]: Unused import

File: Assets/Scripts/ThirdPersonPlayer.cs [hunk 4]
```diff
@@ -2,11 +2,17 @@
...
 
 public class ThirdPersonPlayer : MonoBehaviour
+
 {
+    
+    public Transform cam;
```
Comment (xAmaranth) [inline Assets/Scripts/ThirdPersonPlayer.cs]: cam is unused.

File: Assets/Scripts/ThirdPersonPlayer.cs [hunk 5]
```diff
@@ -2,11 +2,17 @@
...
     public float speed;
+    public float rotateSpeed = 5.0f;
```
Comment (xAmaranth) [inline Assets/Scripts/ThirdPersonPlayer.cs]: rotateSpeed is unused.

File: Assets/Scripts/ThirdPersonPlayer.cs [hunk 6]
```diff
@@ -80,18 +80,27 @@ private void FixedUpdate()
...
                 interpretedYInput = -interpretedYInput;
             }
-            
-            // An upwards rotation (from the Mouse Y Axis) is a rotation about the X Axis.
-            // Similarly, a sideways rotation (from Mouse X) is a rotation about the Y Axis.
-            transform.Rotate(interpretedYInput, interpretedXInput, 0, Space.Self);
+
+            Transform transform1;
+            (transform1 = transform).Rotate(interpretedYInput, interpretedXInput, 0, Space.Self);
+            Quaternion q = transform1.rotation;
```
Comment (xAmaranth) [inline Assets/Scripts/ThirdPersonPlayer.cs]: Use var

File: Assets/Scripts/ThirdPersonPlayer.cs [hunk 7]
```diff
@@ -80,18 +80,27 @@ private void FixedUpdate()
...
+            q.eulerAngles = new Vector3(q.eulerAngles.x, q.eulerAngles.y, 0);
+            transform1.rotation = q;
```
Comment (xAmaranth) [inline Assets/Scripts/ThirdPersonPlayer.cs:89]: Whatever this wizardry is, good job because it works.

**Review by xAmaranth — COMMENTED** [review]
Only minor readability adjustments.

**Comment by irisong980912** [comment]
Does not work for me

**Pull request merged**